### PR TITLE
Fix clang build

### PR DESCRIPTION
**Pull request by vchong**

### CODE REVIEW

**Comment by vchong** [comment]
Depends on https://github.com/jforissier/docker_optee_os_ci_clangbuilt/pull/1.

File: ta/arch/arm/link.mk [hunk 1]
```diff
@@ -49,7 +49,7 @@ cleanfiles += $(link-out-dir$(sm))/dyn_list
 
 link-ldadd  = $(user-ta-ldadd) $(addprefix -L,$(libdirs))
 link-ldadd += --start-group $(addprefix -l,$(libnames)) --end-group
-ldargs-$(user-ta-uuid).elf := $(link-ldflags) $(objs) $(link-ldadd)
+ldargs-$(user-ta-uuid).elf := $(link-ldflags) $(objs) $(link-ldadd) $(libgcc$(sm))
```
Comment (jforissier) [inline ta/arch/arm/link.mk:52]: I think `$(libgcc$(sm))` is needed in `link_shlib.mk` too: https://github.com/OP-TEE/optee_os/blob/3.8.0/ta/arch/arm/link_shlib.mk#L27

File: mk/clang.mk [hunk 2]
```diff
@@ -44,5 +44,5 @@ OBJCOPY		:= false
 OBJDUMP		:= false
 READELF		:= false
 nostdinc	:= --bad-nostdinc-variable
-libgcc  	:= --bad-libgcc-variable
+libgcc		:= --bad-libgcc-variable
```
Comment (jforissier) [inline mk/clang.mk:47]: Unrelated change

File: mk/clang.mk [hunk 3]
```diff
@@ -26,8 +26,8 @@ comp-cflags-warns-clang := -Wno-language-extension-token \
 
 # Note, the currently used compiler runtime library may be libgcc.a or
 # libclang_rt.builtins.*.a depending on the compiler build-time configuration.
-libgcc$(sm)  	:= $(shell $(CC$(sm)) $(CFLAGS$(arch-bits-$(sm))) $(comp-cflags$(sm)) \
-			-print-libgcc-file-name 2> /dev/null)
+libgcc$(sm)	:= $(shell $(CC$(sm)) $(CFLAGS$(arch-bits-$(sm))) $(comp-cflags$(sm)) \
+			-rtlib=compiler-rt -print-libgcc-file-name 2> /dev/null)
```
Comment (jforissier) [inline mk/clang.mk:30]: The comment needs to be updated

**Comment by vchong** [comment]
>> Add missing $libgcc to ldargs for ldelf and ta. This is required for the clang build.

> Why it this required? The code currently builds fine with Clang 9 and 10 at least. You should explain a bit more. Is it related to a particular version or build of Clang?

I get a bunch of `ld.lld: error: undefined symbol: __aeabi_mem*` errors without it using AOSP's prebuilt clang. Might be a bionic vs glibc thing.

**Comment by vchong** [comment]
All comments addressed. Changes are forced pushed due to changes in commit message.

**Comment by jforissier** [comment]
I have restarted Shippable because it looks like it used an outdated Docker image. It should pull the updated one now.

**Comment by vchong** [comment]
Rebased and tags applied. Thanks!
